src/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration/XElementProcessor.cs

```diff
@@ -0,0 +1,231 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Xml.Linq;
+using Microsoft.OpenApi.Any;
+using Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration.Exceptions;
+using Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration.Extensions;
+using Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration.Models.KnownStrings;
+using Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration.ReferenceRegistries;
+using Microsoft.OpenApi.Models;
+using Microsoft.OpenApi.Readers;
+
+namespace Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration
+{
+    /// <summary>
+    /// Holds common functionality to process xelement that are used across various <see cref="IFilter"/>.
+    /// </summary>
+    public static class XElementProcessor
```

Comment:
>public [](start = 4, length = 6)

internal?

Reply:
it might be useful for custom filters that might be written, but i guess can change it afterwards

---
In reply to: [199572380](https://github.com/Microsoft/OpenAPI.NET.CSharpAnnotations/pull/137#discussion_r199572380) [](ancestors = 199572380)